scripts/format.sh

```
@@ -11,4 +11,13 @@ echo "Fixing trailing whitespace and line endings..."
 # Remove trailing whitespace
 find src/ tests/ -name "*.py" -type f -exec sed -i 's/[[:space:]]*$//' {} +
 
+echo "Ensuring all files end with a newline..."
+# Find files that do not end with a newline and add one
+FILES_TO_FIX=$(find src/ tests/ -type f \( -name "*.py" -o -name "*.sh" \) -exec sh -c 'test -n "$(tail -c 1 "{}" | tr -d "\n")"' _ {} \;)
```

Shellcheck (0.11.0)

[warning] Injecting filenames is fragile and insecure. Use parameters.

(SC2156)